crates/trie/src/state.rs

```diff
@@ -38,17 +39,27 @@ impl HashedPostState {
         state: impl IntoIterator<Item = (&'a Address, &'a BundleAccount)>,
     ) -> Self {
         let mut this = Self::default();
-        for (address, account) in state {
-            let hashed_address = keccak256(address);
-            this.accounts.insert(hashed_address, account.info.clone().map(into_reth_acc));
-
-            let hashed_storage = HashedStorage::from_iter(
-                account.status.was_destroyed(),
-                account.storage.iter().map(|(key, value)| {
-                    (keccak256(B256::new(key.to_be_bytes())), value.present_value)
-                }),
-            );
-            this.storages.insert(hashed_address, hashed_storage);
+
+        let changes = state
+            .into_iter()
+            .collect::<Vec<(&'a Address, &'a BundleAccount)>>()
+            .into_par_iter()
+            .map(|(address, account)| {
+                let hashed_address = keccak256(address);
+                let hashed_account = account.info.clone().map(into_reth_acc);
+                let hashed_storage = HashedStorage::from_iter(
+                    account.status.was_destroyed(),
+                    account.storage.iter().map(|(key, value)| {
+                        (keccak256(B256::new(key.to_be_bytes())), value.present_value)
+                    }),
```

Comment:
let's deconstruct `account` so we avoid the clone

Reply:
I'm sorry, I'm trying to adjust it as below:

```diff
diff --git a/crates/trie/src/state.rs b/crates/trie/src/state.rs
index 212d9742b..693dd526a 100644
--- a/crates/trie/src/state.rs
+++ b/crates/trie/src/state.rs
@@ -45,11 +45,12 @@ impl HashedPostState {
             .collect::<Vec<(&'a Address, &'a BundleAccount)>>()
             .into_par_iter()
             .map(|(address, account)| {
+                let BundleAccount { info, status, storage, .. } = account;
                 let hashed_address = keccak256(address);
-                let hashed_account = account.info.clone().map(into_reth_acc);
+                let hashed_account = info.map(into_reth_acc);
                 let hashed_storage = HashedStorage::from_iter(
-                    account.status.was_destroyed(),
-                    account.storage.iter().map(|(key, value)| {
+                    status.was_destroyed(),
+                    storage.iter().map(|(key, value)| {
                         (keccak256(B256::new(key.to_be_bytes())), value.present_value)
                     }),
                 );
```

but it can't compile, error as below:

```bash
error[E0507]: cannot move out of `*info` which is behind a shared reference
    --> crates/trie/src/state.rs:50:38
     |
50   |                 let hashed_account = info.map(into_reth_acc);
     |                                      ^^^^ ------------------ `*info` moved due to this method call
     |                                      |
     |                                      help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
     |                                      move occurs because `*info` has type `std::option::Option<AccountInfo>`, which does not implement the `Copy` trait
     |
note: `std::option::Option::<T>::map` takes ownership of the receiver `self`, which moves `*info`
    --> /root/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:1068:22
     |
1068 |     pub fn map<U, F>(self, f: F) -> Option<U>
     |                      ^^^^
help: you can `clone` the value and consume it, but this might not be your desired behavior
     |
50   |                 let hashed_account = <std::option::Option<AccountInfo> as Clone>::clone(&info).map(into_reth_acc);
     |                                      ++++++++++++++++++++++++++++++++++++++++++++++++++++    +

For more information about this error, try `rustc --explain E0507`.
```

seems also need `clone()`

Reply:
@onbjerg sorry to bother you again, could you please give me some advice on how to avoid using the `clone()`?

Reply:
ah, you're right - since it's an `iter` you need to clone because `into_reth_acc` doesn't take a ref